php/libraries/Module.class.inc

```diff
@@ -62,6 +62,38 @@ abstract class Module extends \LORIS\Router\PrefixRouter
         return $modules;
     }
 
+    /**
+     * Check whether one or more module descriptors are active from the given
+     * database.
+     *
+     * @param \Database $db      an open connection to a database containing a
+     *                           'modules' table
+     * @param array     $modules a list of the interested module names to check
+     *
+     * @return array of module name => boolean (active or not)
+     */
+    static public function checkActiveModules(\Database $db, array $modules): array
+    {
+        $params = array();
+        $values = array();
+        foreach ($modules as $key => $name) {
+            $para          = "n$key";
+            $params[]      = ":$para";
+            $values[$para] = $name;
+        }
+        $list = $db->pselect(
+            "SELECT Name, Active FROM modules WHERE Name IN (" .
+            implode(', ', $params) . ")",
+            $values
+        );
+
+        $checkedModules = [];
+        foreach ($list as $item) {
+            $checkedModules[$item['Name']] = ($item['Active'] == 'Y'? true : false);
+        }
+        return $checkedModules;
```

Comment:
What if `$modules` contains a module name that does not exist? Should the returned array`$checkedModules` contain a key for the invalid module name (with value `false`, obviously) or not? Example: what would `checkActiveModules($db, ['imaging_browser', 'xxxxxx'])` return?

Reply:
I was assuming that the module name should be checked by the implementer to simplify the issue, I will check.